src/lp_sensitivity2.jl

```diff
@@ -423,12 +429,25 @@ function _standard_form_basis(model::Model, std_form)
         if status == MOI.NONBASIC
             status = _convert_nonbasic_status(c_obj.set)
         end
+        col = std_form.columns[c_obj.func]
+        variable_seen[col] = true
         if status != MOI.BASIC
-            col = std_form.columns[c_obj.func]
             variable_status[col] = status
         end
         bound_status[i] = status
     end
+    for (x, col) in std_form.columns
+        if variable_seen[col]
+            continue
+        end
+        # MOI doesn't provide a way of distinguishing between basic and
+        # non-basic variables if they have no bounds. As a heuristic, assume the
+        # variable is non-basic if it is at 0.
+        #
+        # This doesn't cover degenerate problems (basic variable at 0), or
```

Comment:
@geoffleyland found a case where this happened. We could detect when the basis is not square and throw a better error message. The work-around is to add a bound so the variable isn't free.